mi/dataset/parser/flort_dj.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+"""
+@package mi.dataset.parser.flort_dj
+@file mi-dataset/mi/dataset/parser/flort_dj.py
+@author Rene Gelinas
+@brief Parser for the flort_dj dataset driver
+
+This file contains code for the flort_dj parser and code to produce data
+particles. This parser is for recovered data only - it produces a single
+particle for the data recovered from the instrument.
+
+The input file is ASCII. There are two sections of data contained in the
+input file.  The first is a set of header information, and the second is a set
+of hex ascii data with one data sample per line in the file. Each line in the
+header section starts with a '*'. The header lines are simply ignored.
+Each line of sample data produces a single data particle.
+Malformed sensor data records and all header records produce no particles.
+
+Release notes:
+
+Initial Release
+"""
+
+__author__ = 'Rene Gelinas'
+__license__ = 'Apache 2.0'
+
+import calendar
+import re
+
+from mi.core.log import get_logger
+log = get_logger()
+
+from mi.core.common import BaseEnum
+from mi.core.exceptions import UnexpectedDataException
+
+from mi.core.instrument.dataset_data_particle import \
+    DataParticle, \
+    DataParticleKey, \
+    DataParticleValue
+
+from mi.dataset.dataset_parser import SimpleParser
+
+from mi.dataset.parser.common_regexes import \
+    END_OF_LINE_REGEX, ANY_CHARS_REGEX, ASCII_HEX_CHAR_REGEX
+
+# Basic patterns
+
+# Time tuple corresponding to January 1st, 2000
+JAN_1_2000 = (2000, 1, 1, 0, 0, 0, 0, 0, 0)
+
+# regex for identifying start of a header line
+START_HEADER = r'\*'
+# Header data:
+HEADER_PATTERN = START_HEADER    # Header data starts with '*'
+HEADER_PATTERN += ANY_CHARS_REGEX         # followed by text
+HEADER_PATTERN += END_OF_LINE_REGEX         # followed by newline
+HEADER_MATCHER = re.compile(HEADER_PATTERN)
+
+# this list of strings corresponds to the particle parameter names
+# it is used by build_parsed_values.  Named groups in regex must match.
+DATA_PARTICLE_MAP = [
+    'raw_signal_beta',
+    'raw_signal_chl',
+    'raw_signal_cdom',
+    'ctd_time'
+]
+
+# Regex for data from the Endurance array with the FLORT connected to the CTDBP
+# Each data record is in the following format:
+# ttttttccccccppppppvvvvbbbbllllddddssssssss
+# where each character indicates one hex ascii character.
+# First 6 chars: tttttt = Temperature A/D counts
+# Next 6 chars: cccccc = Conductivity A/D counts
+# Next 6 chars: pppppp = pressure A/D counts
+# Next 4 chars: vvvv = temperature compensation A/D counts
+# Next 4 chars: bbbb = Backscatter in counts (FLORT data omitted from output)
+# Next 4 chars: llll = Chlorophyll in counts (FLORT data omitted from output)
+# Next 4 chars: dddd = CDOM in counts (FLORT data omitted from output)
+# Last 8 chars: ssssssss = seconds since January 1, 2000
+# Total of 42 hex characters and line terminator
+
+CTD_FLORT_DATA_REGEX = r'(?:' + ASCII_HEX_CHAR_REGEX + '{6})'
```

Comment:
why is this duplicated instead of just importing from `mi/dataset/parser/ctdbp_cdef.py`?

Reply:
The regex is different between the two files as far as which data are omitted.  However,  the comment was incorrect in flort_dj.py; copy/paste error.  I've updated the comment to reflect the regex.